MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java

```diff
@@ -1014,20 +1045,62 @@ public void forceLocationUpdate(@Nullable Location location) {
    * @param lookAheadUpdate If set to true, the last location's timestamp has to be greater than current timestamp and
    *                        should represent the time at which the animation should actually reach this position,
    *                        cutting out the time interpolation delay.
+   * @deprecated use {@link #forceLocationUpdate(LocationUpdate)} instead
    */
+  @Deprecated
   public void forceLocationUpdate(@Nullable List<Location> locations, boolean lookAheadUpdate) {
     checkActivationState();
     if (locations != null && locations.size() >= 1) {
+      Location targetLocation = locations.get(locations.size() - 1);
+      if (targetLocation == null) {
+        return;
+      }
+      LocationUpdate.Builder builder = new LocationUpdate.Builder()
+        .location(targetLocation)
+        .intermediatePoints(locations.subList(0, locations.size() - 1));
+
+      if (lookAheadUpdate) {
+        long currentTimestamp;
+        long locationTimestamp;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+          currentTimestamp = SystemClock.elapsedRealtimeNanos();
+          locationTimestamp = targetLocation.getElapsedRealtimeNanos();
+          if (locationTimestamp == 0) {
+            // fallback in case the elapsed realtime is not set
+            currentTimestamp = System.currentTimeMillis();
+            locationTimestamp = targetLocation.getTime();
+          }
+        } else {
+          currentTimestamp = System.currentTimeMillis();
```

Comment:
Here it's millis while above it is set as nanos.
Seems a bit strange to me. I'd rather stick to single time unit and moreover name variables like `currentTimestampNs`.

Reply:
Cleaned up :+1: